### ISSUE TITLE
Clarify implicit version automation

### ISSUE THREAD

**Issue by astrojuanlu** [readthedocs/readthedocs.org#8476]
Some thoughts from #8473, also related to #5319, #8015, and #8460.

In our [Versioned Documentation docs](https://docs.readthedocs.io/en/stable/versions.html) we say:

- That tags _should_ be "numbered in a way that is consistent with semantic versioning", but we don't say what happens otherwise. Having this "how we envision" section feels weird, we should be more descriptive and rephrase this advice somehow. Partially addressed in #8473:

```diff
 we will create a ``latest`` version.
 This will point at the default branch for your VCS control: ``master``, ``default``, or ``trunk``.
 
-We also create a ``stable`` version,
-if your project has any tagged releases.
-``stable`` will be automatically kept up to date to point at your highest version.
+If your project has any tags or branches with a name following `semantic versioning <https://semver.org/>`_,
+we also create a ``stable`` version, tracking your most recent release.
 If you want a custom ``stable`` version,
 create either a tag or branch in your project with that name.
 
```

- Next, that "we in fact are parsing your tag names against the rules given by PEP 440" ❓ Which forces users to understand [what are the differences between PEP 440 and SemVer](https://www.python.org/dev/peps/pep-0440/#semantic-versioning). This however reflects the truth more closely:

https://github.com/readthedocs/readthedocs.org/blob/b68a92106b5926c7e708f77aa3d04e2f30b0a24b/readthedocs/projects/version_handling.py#L36-L40

- And finally, we are _not_ saying that, in fact, we also accept `.x` as a patch segment:

https://github.com/readthedocs/readthedocs.org/blob/b68a92106b5926c7e708f77aa3d04e2f30b0a24b/readthedocs/projects/version_handling.py#L41-L44

This issue concerns updating the documentation to reflect the current reality, although I support continuing the discussion in #5319 about making these Automation Rules that are present for all newly created projects, and that can be controlled.

**Comment by ericholscher** [comment]
I'd like to remove any additional magic that we can. In particular, removing the auto-activating of versions seems like something that Automation Rules are better for, and we should document that. 

I'd love to also get rid of the `stable` code, but I don't believe we've built them into Automation Rules yet.

**Comment by humitos** [comment]
I agree with @ericholscher here. The more magic we can remove, the better. In particular, if we can migrate that behavior to be explicitly configured by the user.

> I'd love to also get rid of the `stable` code, but I don't believe we've built them into Automation Rules yet.

We have automation rules to detect SemVer (on branches/tags) but we haven't built yet the version aliases (#5318), which is the missing part required to have the same `stable` behavior via automation rules.
